lib/Sema/CSDiagnostics.cpp

```diff
@@ -7986,7 +7984,14 @@ bool NoopCheckedCast::diagnoseIfExpr() const {
   if (!expr)
     return false;
 
-  emitDiagnostic(diag::isa_is_always_true, "is");
+  const auto toType = getToType();
+  const auto *cls = toType->getAs<ClassType>();
+  if (getFromType()->isExistentialType() && cls &&
+      cls->getDecl()->getForeignClassKind() == ClassDecl::ForeignKind::CFType) {
```

Comment:
Let's add a tailored fix+diagnostic for this case and name it appropriately to avoid duplicate check here and in CSSimplify. 

Also please adjust the name from `diagnoseIfExpr` to `diagnoseIsExpr` :)